src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceTest.java

```diff
@@ -53,20 +55,21 @@ void should_succeed_and_find_a_task_by_id() {
 
```

Comment:
**cftTaskDatabaseService.findByIdAndObtainPessimisticWriteLock(taskResource.getTaskId());**
If the task is locked and and another resource tries to obtain the lock on that task, is that handled elegantly and should  we include a test for that edge case (as per AC2)